selectrum.el

```diff
@@ -478,6 +491,22 @@ Passed to various hook functions.")
                                  candidate
                                  'face 'selectrum-current-candidate)))
               (insert "\n" candidate)
+              (let ((right-margin
+                     (get-text-property
+                      0 'selectrum-candidate-display-right-margin
+                      candidate)))
+                (when right-margin
```

Comment:
How about `when-let` instead?

Reply:
I used `when-let` before, and it makes CI fail on Emacs 25. I think `when-let` may not exist before 26 or its behavior is not the same.

Reply:
I refactored the code here a little bit. This is the logic before:

1. calculate displayed candidate
2. display the candidate
3. search for right margin property
4. display the right margin text

The problem is we don't have access to the unprocessed candidate after the first step. So if a prefix is concatenated with it, we can't get the right margin property from the beginning of the string. I wrote it this way:

1. get the original candidate
2. calculate the displayed candidate, and get the right margin property
3. display the "displayed candidate"
4. display the right margin text

Reply:
Ok, seems reasonable.